scripts/generator/graphql_generator/csharp/SDK/UnityLoader.cs.erb

```diff
@@ -57,46 +46,32 @@ namespace <%= namespace %>.SDK {
     /// <summary>
     /// Performs network communication to send GraphQL queries between Unity and a Shopify GraphQL endpoint.
     /// </summary>
-    public class UnityLoader : ILoader {
-        /// <summary>
-        /// Shopify store domain where the GraphQL endpoint lives.
-        /// </summary>
-        public string Domain {
-            get {
-                return _Domain;
-            }
-        }
-
-        /// <summary>
-        /// Access token used to communicate with the Shopify Store GraphQL endpoint.
-        /// </summary>
-        public string AccessToken {
-            get {
-                return _AccessToken;
-            }
-        }
-
-        string _Domain;
-        string _AccessToken;
+    public class UnityLoader : BaseLoader {
+        private Dictionary<string,string> _Headers = new Dictionary<string,string>();
 
         /// <param name="domain">Shopify store domain that you'd like to query on</param>
         /// <param name="accessToken">access token used to communicate with the Shopify Store</param>
-        public UnityLoader(string domain, string accessToken) {
-            _AccessToken = accessToken;
-            _Domain = domain;
-        }
+        public UnityLoader(string domain, string accessToken) : base(domain, accessToken) {}
 
         /// <summary>
         /// Send a query to the Storefront API.
         /// </summary>
         /// <param name="query">Query that will be sent to the Storefront API</param>
         /// <param name="callback">callback that receives a response</param>
-        public void Load(string query, LoaderResponseHandler callback) {
+        override public void Load(string query, LoaderResponseHandler callback) {
             GlobalGameObject.AddComponent<LoaderComponent>()
             .SetURL("https://" + Domain + "/api/graphql.json")
-            .SetAccessToken(AccessToken)
+            .SetHeaders(_Headers)
             .Load(query, callback);
         }
+
+        override public void SetHeader(string key, string value) {
+            _Headers[key] = value;
+        }
+
+        override public string SDKVariantName() {
+            return "unity";
```

Comment:
I think this Variant Name now needs to be generated so we can track whether the code was distributed via Github/Asset Store

Reply:
This was used to identify the engine code before and we use the `ShopifyBuy.VERSION` to indicate which version/distribution of the codebase we use (`github`/`asset-store`). Do we want to keep this as the engine identifier or change it to also identify the distro?

Reply:
Ah yes. At one time we were using the SDK variant name to differentiate and I decided later on that version would be a better place to do this.